lilypad/developer-resources/create-lilypad-module/building-your-module.md

```diff
@@ -0,0 +1,349 @@
+```
+
+You're almost ready to download the model. All you need to do now is replace the following 2 lines:
+
+```python
+tokenizer = AutoTokenizer.from_pretrained(MODEL_IDENTIFIER)
+model = AutoModelForSequenceClassification.from_pretrained(MODEL_IDENTIFIER)
+```
+
+Instead of using `AutoTokenizer` and `AutoModelForSequenceClassification`, replace those with the `DistilBertTokenizer` and `DistilBertForSequenceClassification` we imported.
+
+```python
+tokenizer = DistilBertTokenizer.from_pretrained(MODEL_IDENTIFIER)
+model = DistilBertForSequenceClassification.from_pretrained(MODEL_IDENTIFIER)
+```
+
+The script is now configured! Try running the command again.
+
+```sh
+python -m scripts.download_models
+```
+
+The `models` directory should now appear in your project. 🎉
+
+{% hint style="info" %}
+No matter which model you are using, be sure to thoroughly read the model's documentation to learn how to properly download and use the model locally.
+{% endhint %}
+
+## Preparing Your Module
+
+Before you are able to run your module, we need to build the Docker image. You can run the following command:
+
+```sh
+python -m scripts.docker_build
+```
+
+You should see the following response in the console:
+
+```sh
+❌ Error: DOCKER_REPO is not set in config/constants.py.
+```
+
+Open the `constants.py` file, it should look like this:
+
+```python
+# TODO: Set the Docker Hub repository before pushing the image.
+# Example: "devlinrocha/lilypad-module-sentiment"
+DOCKER_REPO = ""
+
+# TODO: Set the tag for the Docker image.
+# Example: "latest", "v1.0", or a commit SHA
+DOCKER_TAG = "latest"
+
+# TODO: Set the GitHub repository URL where your module is stored.
+# Example: "github.com/devlinrocha/lilypad-module-sentiment".
+MODULE_REPO = ""
+
+# TODO: Specify the target branch name or commit hash.
+# Example: "main" or "c3ed392c11060337cae010862b1af160cd805e67"
+TARGET_COMMIT = "main"
+```
+
+For now, we'll be testing the module locally, so all we need to worry about is the `DOCKER_REPO` variable. We'll use `MODULE_REPO` when it's time to run the module on Lilypad Network. For help or more information, view the [configuration documentation](./configuration.md)
+
+You should be able to successfully build the Docker image now.
+
+{% hint style="info" %}
+In the modules Dockerfile, you'll find 3 COPY instructions.
+
+```Dockerfile
+COPY requirements.txt .
+COPY src /src
+COPY models /models
+```
+
+These instructions bring the `requirements.txt` file, the `src` directory, and the `models` directory into the Docker image. It's important to remember that any modifications to these files or directories will necessitate a rebuild of the module's Docker image to ensure the changes are reflected in the container.
+{% endhint %}
+
+## Building Your Model
+
+Now for the fun part, it's time to start using the model!
+
+This time we'll get started by running the `run_module` script.
+
+```sh
+python -m scripts.run_module
```

Comment:
This will only work if Lilypad CLI is installed on the users machine. Refer to comment in prerequisites